src/GreenPipes/IContextSource.cs

```diff
@@ -0,0 +1,10 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+
+namespace GreenPipes
+{
+    public interface IContextSource<TContext> where TContext : class, PipeContext
+    {
+        Task Fill(IPipe<TContext> pipe, CancellationToken token = default(CancellationToken));

```

Comment:
I don't like the TPL `LinkTo` - wondering if we should stick with the `Connect` that is already part of the vocabulary for the dispatch pipe.

Also, pipes could be connected by observers PostCreate, so that it's ready to rock on Start.